VideoLocker/src/main/java/org/edx/mobile/http/OauthHeaderRequestInterceptor.java

```diff
@@ -1,44 +1,44 @@
 package org.edx.mobile.http;
 
 import android.content.Context;
+import android.support.annotation.NonNull;
 
 import org.edx.mobile.logger.Logger;
 import org.edx.mobile.authentication.AuthResponse;
+import org.edx.mobile.module.prefs.LoginPrefs;
 import org.edx.mobile.module.prefs.PrefManager;
 
 import java.io.IOException;
 
 import okhttp3.Interceptor;
 import okhttp3.Request;
 import okhttp3.Response;
+import roboguice.RoboGuice;
 
 /**
  *  this interceptor inject oauth token to request header
  **/
 public final class OauthHeaderRequestInterceptor implements Interceptor {
     protected final Logger logger = new Logger(getClass().getName());
-    private Context context;
-    public OauthHeaderRequestInterceptor(Context context){
+    @NonNull
+    private final Context context;
+
+    @NonNull
+    private final LoginPrefs loginPrefs;
+
+    public OauthHeaderRequestInterceptor(@NonNull Context context){
         this.context = context;
+        loginPrefs = RoboGuice.getInjector(context).getInstance(LoginPrefs.class);
     }
     @Override public Response intercept(Chain chain) throws IOException {
         Request originalRequest = chain.request();
 
         Request.Builder builder = originalRequest.newBuilder();
-        // generate auth headers
-        PrefManager pref = new PrefManager(context, PrefManager.Pref.LOGIN);
-        AuthResponse auth = pref.getCurrentAuth();
-
-        if (auth == null || !auth.isSuccess()) {
-            // this might be a login with Facebook or Google
-            String token = pref.getString(PrefManager.Key.AUTH_TOKEN_SOCIAL);
-            if (token != null) {
-                builder.addHeader("Authorization", token);
-            } else {
-                logger.warn("Token cannot be null when AUTH_JSON is also null, something is WRONG!");
-            }
+        final String token = loginPrefs.getAuthorizationHeader();
+        if (token != null) {
+            builder.addHeader("Authorization", token);
         } else {
-            builder.addHeader("Authorization", String.format("%s %s", auth.token_type, auth.access_token));
+            logger.warn("Token cannot be null when AUTH_JSON is also null, something is WRONG!");
```

Comment:
MAybe this is out of scope:

I always see this in the log which makes it seem like there is always something wrong. Is this supposed to always come up as a warning?